### ISSUE TITLE
RCS1244 false positive

### ISSUE THREAD

**Issue by hankovich** [JosefPihrt/Roslynator#784]
**Product and Version Used**: 
Roslynator.Analyzers V3.1.0

**Steps to Reproduce**:
```csharp
var ints = Enumerable.Empty<int>();

var array = ints.Select(i =>
{
    return default(int);
}).ToArray();
```

**Actual Behavior**:
RCS1244 is reported and the fix (change `default(int)` to `default`) is suggested.
After applying the proposed fix code is not compiling anymore.

**Expected Behavior**:
No dignostics are reported.
